src/Services/FHIR/FhirValueSetService.php

```diff
@@ -0,0 +1,184 @@
+     *  | address-types                   | both              | Postal & Physical     |  30 |          0 |            0 |         | NULL  |       |
+     *  | address-types                   | physical          | Physical              |  20 |          0 |            0 |         | NULL  |       |
+     *  | address-types                   | postal            | Postal                |  10 |          0 |            0 |         | NULL  |       |
+     *  | address-uses                    | billing           | Billing               |  50 |          0 |            0 |         | NULL  |       |
+     *  | address-uses                    | home              | Home                  |  10 |          0 |            0 |         | NULL  |       |
+     *  | address-uses                    | old               | Old/Incorrect         |  40 |          0 |            0 |         | NULL  |       |
+     *  | address-uses                    | temp              | Temporary             |  30 |          0 |            0 |         | NULL  |       |
+     *  | address-uses                    | work              | Work                  |  20 |          0 |            0 |         | NULL  |       |
+     *  | adjreason                       | Adm adjust        | Adm adjust            |   5 |          0 |            1 |         | NULL  |       |
+     *  | adjreason                       | After hrs calls   | After hrs calls       |  10 |          0 |            1 |         | NULL  |       |
+     *  | adjreason                       | Bad check         | Bad check             |  15 |          0 |            1 |         | NULL  |       |
+     *  | adjreason                       | Bad debt          | Bad debt              |  20 |          0 |            1 |         | NULL  |       |
+     *  | adjreason                       | Coll w/o          | Coll w/o              |  25 |          0 |            1 |         | NULL  |       |
+     *
+     */
+
+
+
+    const USCGI_PROFILE_URI = 'http://hl7.org/fhir/StructureDefinition/shareablevalueset';
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->appointmentService = new AppointmentService();
+        $this->listOptionService = new ListService();
+    }
+
+    /**
+     * Returns an array mapping FHIR Resource search parameters to OpenEMR search parameters
+     */
+    protected function loadSearchParameters()
+    {
+        return [
+        '_id' => new FhirSearchParameterDefinition('_id', SearchFieldType::TOKEN, [new ServiceField('uuid', ServiceField::TYPE_UUID)]),
+        ];
+    }
+
+    /**
+     * Retrieves all of the fhir observation resources mapped to the underlying openemr data elements.
+     * @param $fhirSearchParameters The FHIR resource search parameters
+     * @return processing result
+     */
+    public function getAll($fhirSearchParameters, $puuidBind = null): ProcessingResult
```

Comment:
You need to handle the $fhirSearchParameters here in the valueset for the ```_id``` search parameter.  Otherwise the FhirServiceBase::getOne() call which is being called in your FhirValueSetRestController() is just going to return the entire list of values here with no filtering. on the search parameters. 

Since you are handling this in memory you'll just have to key off the calendar category or the list_id that matches with the _id search parameter.  Remember in FHIR that you can specify multiple ```_id``` values in a search query so the search field will give you an array of ids (usually this will just be one, but again it can be multiples).

Let me know if you need more clarity on this.